File: repos_estudiantes_ALVAREZ_tercera_entrega/CHINO_BLANCO_NOEMI/Generacidad/Ejercicio2/Catalogo.java
package Generacidad.Ejercicio2;


import java.util.ArrayList;

public class Catalogo<T> {
    private ArrayList<T> elementos;

    public Catalogo() {
        elementos = new ArrayList<>();
    }

    public void agregar(T elemento) {
        elementos.add(elemento);
    }

    public T buscar(String criterio) {
        for (T e : elementos) {
            if (e.toString().contains(criterio)) {
                return e;
            }
        }
        return null;
    }

    public void mostrarCatalogo() {
        for (T e : elementos) {
            System.out.println(e);
        }
    }

    public static void main(String[] args) {
        Catalogo<String> catalogoLibros = new Catalogo<>();
        catalogoLibros.agregar("Adela zamudio");
        catalogoLibros.agregar("LA VACA");
        System.out.println(" Catálogo de Libros:");
        catalogoLibros.mostrarCatalogo();

        Catalogo<String> catalogoProductos = new Catalogo<>();
        catalogoProductos.agregar("Radio");
        catalogoProductos.agregar("antena");
        System.out.println(" Catálogo de Productos:");
        System.out.println("noemi chino blanco");
        catalogoProductos.mostrarCatalogo();
    }
}
 
    

File: repos_estudiantes_ALVAREZ_tercera_entrega/CHINO_BLANCO_NOEMI/Generacidad/Ejercicio2/Catalogo.py
class Catalogo:
    def __init__(self):
        self.elementos = []

    def agregar(self, elemento):
        self.elementos.append(elemento)

    def buscar(self, criterio):
        for elemento in self.elementos:
            if criterio in str(elemento):
                return elemento
        return None

    def mostrar_catalogo(self):
        for elemento in self.elementos:
            print(elemento)

catalogo_libros = Catalogo()
catalogo_libros.agregar("Adela zamudio")
catalogo_libros.agregar("LA VACA")
print(" Catálogo de Libros:")
catalogo_libros.mostrar_catalogo()

catalogo_productos = Catalogo()
catalogo_productos.agregar("radio")
catalogo_productos.agregar("antena")
print(" Catálogo de Productos:")
print("noemi chino blanco")
catalogo_productos.mostrar_catalogo()
